CONTRIBUTING.md

```diff
@@ -0,0 +1,196 @@
+
+- ##### Loading / saving logs
+
+  It can save all the collected logs that are collected in the background to a JSON file. To save logs, the `saveLogs` instruction from [`save-load.js`](https://github.com/mozilla/extension-activity-monitor/blob/68d51940f1db397a0972658622bbdd39041436a7/src/lib/save-load.js#L22-L27) is sent to the background where the save operation occurs.
+
+  The saved logs can be viewed by loading the JSON file. It will read the logs and create a new tab. The tab id along with the loaded logs are [saved in the Background](https://github.com/mozilla/extension-activity-monitor/blob/68d51940f1db397a0972658622bbdd39041436a7/src/lib/ext-monitor.js#L6). The Activity Log page initializes in the new tab by fetching logs from background with it's tab id and render them in a [`log-view`](https://github.com/mozilla/extension-activity-monitor/blob/master/src/lib/web-component/log-view/). When the tab is closed, the loaded logs along with the tab id will be removed from Background.
+
+#### Rendering Logs
+
+The logs are being rendered using [`log-view`](https://github.com/mozilla/extension-activity-monitor/blob/master/src/lib/web-component/log-view/) web component. While filters are being applied, any new log that doesn't match the applied filters will be rendered as hidden in [`log-view`](https://github.com/mozilla/extension-activity-monitor/blob/master/src/lib/web-component/log-view/).
+
+#### Filtering Log Entries
+
+Extension Activity Monitor offers the following filtering options-
+
+- Filter logs with extension id, view type, API name, API type. Whenever a new unknown identity is found in new logs, that is being added as a checkbox (checked by default) in the **View**.
+  - These filtering options use the [`filter-option`](https://github.com/mozilla/extension-activity-monitor/tree/master/src/lib/web-component/filter-option) web component.
+- Filter logs with substring.
+  - It uses the [`filter-keyword`](https://github.com/mozilla/extension-activity-monitor/tree/master/src/lib/web-component/filter-keyword) web component.
+- Filter logs with range of timestamp. It is implemented with the help of "context menu" in `log-view`.
+  - It uses the [`filter-timestamp`](https://github.com/mozilla/extension-activity-monitor/tree/master/src/lib/web-component/filter-timestamp) web component.
+- Filter logs with tab id.
+  - When Activity Log page is opened through devtools panel, logs will be filtered by that tab's id.
```

Comment:
This is almost user documentation (short of the links to the files). Can we focus more on the architecture/components here?

Reply:
Should I remove the following part- 

> 
> - Filter logs with extension id, view type, API name, API type. Whenever a new unknown identity is found in new logs, that is being added as a checkbox (checked by default) in the **View**.
>  - These filtering options use the [`filter-option`](https://github.com/mozilla/extension-activity-monitor/tree/master/src/lib/web-component/filter-option) web component.
> - Filter logs with substring.
>  - It uses the [`filter-keyword`](https://github.com/mozilla/extension-activity-monitor/tree/master/src/lib/web-component/filter-keyword) web component.
> - Filter logs with range of timestamp. It is implemented with the help of "context menu" in `log-view`.
>  - It uses the [`filter-timestamp`](https://github.com/mozilla/extension-activity-monitor/tree/master/src/lib/web-component/filter-timestamp) web component.
> - Filter logs with tab id.
>  - When Activity Log page is opened through devtools panel, logs will be filtered by that tab's id.